filebeat/fileset/config.go

```diff
@@ -54,3 +56,21 @@ func NewFilesetConfig(cfg *common.Config) (*FilesetConfig, error) {
 
 	return &fcfg, nil
 }
+
+// mergePathDefaults returns a copy of c containing the path variables that must
+// be available for variable expansion in module configuration (e.g. it enables
+// the use of ${path.config} in module config).
+func mergePathDefaults(c *common.Config) (*common.Config, error) {
+	defaults := common.MustNewConfigFrom(map[string]interface{}{
+		"path": map[string]interface{}{
+			"home":   paths.Paths.Home,
+			"config": "${path.home}",
+			"data":   filepath.Join("${path.home}", "data"),
+			"logs":   filepath.Join("${path.home}", "logs"),
+		},
+	})
+	if err := defaults.Merge(c); err != nil {
+		return nil, err
+	}
+	return defaults, nil
+}
```

Comment:
This function suggest that module configurations can overwrite the `path.X` settings. Not sure this is semantically correct. The `paths` settings are somewhat special, in being global settings. They can be changed via CLI flags, or via global settings in the beat configuration. Being a set of beats global setting, the settings should appear as 'readonly' to modules, and not be overwritable.

Reply:
In the original code the merging was in reverse order, such that modules trying to overwrite the settings would not be able to do so. The code used to overwrite with the known values always.

The change in the PR reverses order, such that we first merge the correct values and merge the module settings. This is a change in semantics, as modules will be able to overwrite paths now.